openedx/core/djangoapps/discussions/api/config.py

```diff
@@ -0,0 +1,74 @@
+from typing import Iterable, List, Optional
+
+from django.contrib.sites.models import Site
+from django.db.models import Q
+from opaque_keys.edx.keys import CourseKey
+
+# TODO DiscussionProviderConfig
+# TODO DiscussionPluginConfigData
+
+from .data import CourseDiscussionConfigData, DiscussionPluginConfigData
+from ..models import LearningContextDiscussionConfig
+from ...config_model_utils.models import site_from_org
+
+
+def _org_and_site_from_course_key(context_key: CourseKey):
+    org_key = getattr(context_key, 'org', None)
+    site = site_from_org(org_key)
+    return org_key, site
+
+
+def get_course_discussion_config(course_key: CourseKey) -> Optional[CourseDiscussionConfigData]:
+    """
+    Returns the active discussion configuration for the course.
+
+    Args:
+        course_key (CourseKey): Learning context, currently only a course
+
+    Returns:
+        A :class:`CourseDiscussionConfigData` object with the active configuration for this course.
+        Returns `None` if a discussion tool isn't configured for the course yet.
+
+    """
+    try:
+        course_config = LearningContextDiscussionConfig.objects.get(pk=course_key)
+    except LearningContextDiscussionConfig.DoesNotExist:
+        return None
+    return CourseDiscussionConfigData(
```

Comment:
Oh - is this a pattern we do?  Pulling data out of a model and putting it into a `*Data` object?  My question elsewhere about LearningContextDiscussionConfig and CourseDiscussionConfigData being the same thing is clearly wrong, then... It does feel to me like the names should use the same nouns, though.

Reply:
Yes, that was a new API being developed from scratch so I decided to based this on latest standards being used int eh codebase. Here is the relevant [ADR](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/learning_sequences/docs/decisions/0001-extensions-to-inter-app-apis.rst)